src/main/ngapp/accio/src/app/editor/magic-wand.service.ts

```diff
@@ -7,95 +7,116 @@ import { SetOperator } from './set-operator';
-      // Add coords of adjacent pixels to the heap
+      // Adds coords of adjacent pixels to the heap.
       for (const neighborPixel of neighbors) {
         const x: number = neighborPixel[0];
         const y: number = neighborPixel[1];
-        // Check if coord is in bounds and has not been visited first
+        // Checks if coord is in bounds and has not been visited first.
         if (!this.getIsValid(imgData.width, imgData.height, x, y, visited)) {
           continue;
         }
-        // Visit the pixel and check if it should be part of the mask
+        // Visits the pixel and checks if it should be part of the mask.
         visited.add(this.coordToDataArrayIndex(x, y, imgData.width));
         if (this.getIsMask(originalPixel, imgData, neighborPixel, tolerance)) {
           visit.push(neighborPixel);
         }
       }
-    }  // end of while loop
+    }  // End of while loop...
 
     return mask;
   }
 
-  /**Judge current pixel's RGBA against original pixel's RGBA to
+  /**Judges current pixel's RGB against original pixel's RGB to
    * see if it can still be part of the mask (using tolerance criteria).
    */
   getIsMask(originalPixel: Array<number>, imgData: ImageData,
       pixelCoord: Array<number>, tolerance: number): boolean {
     const curX: number = pixelCoord[0];
     const curY: number = pixelCoord[1];
 
-    // Get array of attributes of current pixel
-    const curPixel: Array<number> = this.dataArrayToRgba(imgData, curX, curY);
+    // Gets [R, G, B] of current pixel.
+    const curPixel: Array<number> = this.dataArrayToRgb(imgData, curX, curY);
 
-    // All attributes of the pixel (R,G,B, and A) must be within tolerance level
-    for (let i = 0; i < 4; i++) {
-      const upperTolerance: boolean = curPixel[i] > originalPixel[i] + tolerance;
-      const lowerTolerance: boolean = curPixel[i] < originalPixel[i] - tolerance;
-      if (upperTolerance || lowerTolerance) {
-        return false;
-      }
+    const colorDifference = this.rgbEuclideanDist(originalPixel, curPixel);
+
+    if (colorDifference > tolerance) {
+      return false;
     }
 
     return true;
   }
 
-  // Check if @pixelCoord is in bounds and makes sure it's not a repeat coord
+  /**@returns {number} the straight line distance between the two colors
+   * @param {Array<number> [R, G, B]} basisColor and 
+   * @param {Array<number> [R, G, B]} secondColor
+   */
+  rgbEuclideanDist(basisColor: Array<number>, secondColor: Array<number>)
+      : number {
+    if (basisColor.length != secondColor.length) {
+      throw new Error(
+          'basisColor and secondColor must be same lengthed arrays...');
+    }
+    if (basisColor.length != 3) {
+      throw new Error(
+          'Arguments must be an array of [R, G, B] (length == 3)...');
+    }
+
+    let distance = 0;
+
+    for (let i = 0; i < basisColor.length; i++) {
+      distance += Math.pow((basisColor[i] - secondColor[i]), 2);
+    }
+
+    return Math.sqrt(distance);
```

Comment:
sqrt is an expensive operation so I would suggest just working in the squared space.  So if you square your tolerance you can accomplish the same thing.